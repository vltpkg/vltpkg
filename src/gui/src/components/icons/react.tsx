export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
