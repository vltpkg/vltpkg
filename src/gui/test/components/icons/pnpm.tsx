import { afterEach, expect, test } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { Pnpm } from '@/components/icons/index.js'

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  cleanup()
})

test('pnpm icon render default', () => {
  render(<Pnpm />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('pnpm icon render custom class', () => {
  render(<Pnpm className="custom-class" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})
