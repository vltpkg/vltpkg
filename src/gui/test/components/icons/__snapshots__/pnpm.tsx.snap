// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pnpm icon render custom class 1`] = `

<div>
  <svg
    class="custom-class"
    id="a"
    xmlns="http://www.w3.org/2000/svg"
    width="256"
    height="255.98"
    viewbox="0 0 256 255.98"
  >
    <path
      id="b"
      d="M256,79.99h-79.99V0h79.99V79.99h0Z"
    >
    </path>
    <path
      id="c"
      d="M167.99,79.99H88V0h79.99V79.99Z"
    >
    </path>
    <path
      id="d"
      d="M80,79.99H0V0H80V79.99Z"
    >
    </path>
    <path
      id="e"
      d="M256,167.99h-79.99V87.99h79.99v80h0Z"
    >
    </path>
    <path
      id="f"
      d="M167.99,167.99H88V87.99h79.99v80Z"
    >
    </path>
    <path
      id="g"
      d="M167.99,255.98H88v-79.99h79.99v79.99Z"
    >
    </path>
    <path
      id="h"
      d="M256,255.98h-79.99v-79.99h79.99v79.99h0Z"
    >
    </path>
    <path
      id="i"
      d="M80,255.98H0v-79.99H80v79.99Z"
    >
    </path>
  </svg>
</div>

`;

exports[`pnpm icon render default 1`] = `

<div>
  <svg
    id="a"
    xmlns="http://www.w3.org/2000/svg"
    width="256"
    height="255.98"
    viewbox="0 0 256 255.98"
  >
    <path
      id="b"
      d="M256,79.99h-79.99V0h79.99V79.99h0Z"
    >
    </path>
    <path
      id="c"
      d="M167.99,79.99H88V0h79.99V79.99Z"
    >
    </path>
    <path
      id="d"
      d="M80,79.99H0V0H80V79.99Z"
    >
    </path>
    <path
      id="e"
      d="M256,167.99h-79.99V87.99h79.99v80h0Z"
    >
    </path>
    <path
      id="f"
      d="M167.99,167.99H88V87.99h79.99v80Z"
    >
    </path>
    <path
      id="g"
      d="M167.99,255.98H88v-79.99h79.99v79.99Z"
    >
    </path>
    <path
      id="h"
      d="M256,255.98h-79.99v-79.99h79.99v79.99h0Z"
    >
    </path>
    <path
      id="i"
      d="M80,255.98H0v-79.99H80v79.99Z"
    >
    </path>
  </svg>
</div>

`;
